### ISSUE TITLE
Why blockchain.currentBlock height is 1 after reboot geth(height 100 before)?

### ISSUE THREAD

**Issue by friends110110** [ethereum/go-ethereum#18482]
Sometimes, why the current block height is 1 when the 'geth' reboot ( height is 100 before), so it tooltips `ErrKnownBlock = errors.New("block already known")`
If the geth reboot, why the block height is 1 or less than 100 (which before reboot)?
```
// ValidateBody validates the given block's uncles and verifies the block
// header's transaction and uncle roots. The headers are assumed to be already
// validated at this point.
func (v *BlockValidator) ValidateBody(block *types.Block) error {
	// Check whether the block's known, and if not, that it's linkable
	if v.bc.HasBlockAndState(block.Hash(), block.NumberU64()) {
		return ErrKnownBlock
	}
.....
```

**Comment by sourav1547** [comment]
I have the same question. Did you manage to find an answer?

**Comment by pchain-org** [comment]
found one possible reason. Look into 

    func  (bc *BlockChain) writeBlockWithState() {
        ...
        //restart here???
        if status == CanonStatTy {
		bc.insert(block)
	    }
        ...
    }
   
if geth restart just before refresh the head block hash, this issue could happen

**Comment by rjl493456442** [comment]
It's fixed by #19417
